azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -1053,6 +1087,24 @@ func flattenKubernetesClusterLinuxProfile(profile *containerservice.LinuxProfile
 	return []interface{}{values}
 }
 
+func flattenKubernetesClusterWindowsProfile(profile *containerservice.ManagedClusterWindowsProfile) []interface{} {
+	if profile == nil {
+		return []interface{}{}
+	}
+
+	values := make(map[string]interface{})
+
+	if username := profile.AdminUsername; username != nil {
+		values["admin_username"] = *username
+	}
+
+	if password := profile.AdminPassword; password != nil {
+		values["admin_password"] = *password
+	}
```

Comment:
@titilambert as you've mentioned unfortunately Azure doesn't return this information; as such we'll need to try and load that from the config.

You can do that by passing `d` in from the Read method, and then we can try and obtain it like so:

```
if nodePoolsRaw, ok := d.GetOk("agent_profile_pool"); ok {
  nodePools := nodePoolsRaw.([]interface{})
  for _, poolRaw := range nodePools {
    // handles someone running `-apply=foo.bar[2]` where only the second element's non-nil
    if poolRaw == nil {
      continue
    }
    pool := poolRaw.(map[string]interface{})

    // ensure the pool name matches
    currentPoolName := pool["name"].(string)
    if poolName != currentPoolName {
      continue
    }

    // then try and obtain this from the config
    if windowsProfilesRaw, ok := pool["windows_profile"]; ok {
      windowsProfiles := windowsProfilesRaw.([]interface{})
      if len(windowsProfiles) == 0 {
        continue
      }

      windowsProfile:= windowsProfiles[0].(map[string]interface{}
      values["admin_password"] = windowsProfile["admin_password"]
      break
    }
  }
}
```

Whilst the configuration may exist in the code (and thus the state) it's not necessarily guaranteed (for example, whilst it should exist during Create and Update - it may not exist at Import time, and so this'll be nil/empty, which we need to guard against)

🤞 that should work :)

Reply:
Thanks for the input, I think that @titilambert got the `plan` on an existing cluster to pass without issue now!